packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -56,8 +61,22 @@ export default function LinkPreview( {
 
 	let icon;
 
+	const attachmentImg =
+		value.type === 'attachment' && isImageUrl( value?.url )
+			? value.url
+			: null;
+
 	if ( richData?.icon ) {
 		icon = <img src={ richData?.icon } alt="" />;
+	} else if ( attachmentImg ) {
+		//TODO we don't have an alt text for this image
+		icon = (
+			<img
+				className="block-editor-link-control__search-item-media-thumbnail"
+				src={ attachmentImg }
```

Comment:
this loads the original sized image, it feels a little overkill

Reply:
How big is the image? If it's huge then I don't think this is acceptable.

I did have a plan to use the `value.id` to fetch the full data about the entity from the Core Store. In fact I did have a PR about precisely that but I cannot locate it.

This selector would need to be passed in via blockEditorSettings because - as you noted - we cannot consume Core Store directly here.

If we do this I would consider whether it's better to do that bit as a separate PR.

Reply:
https://github.com/WordPress/gutenberg/pull/50998 this one perhaps?